avocado/core/status/repo.py

```diff
@@ -45,6 +45,10 @@ def get_task_data(self, task_id):
         """Returns all data on a given task, by its ID."""
         return self._all_data.get(task_id)
 
+    def get_latest_task_data(self, task_id):
+        """Returns the latest data on a given task, by its ID."""
+        return self._all_data.get(task_id)[-1]
```

Comment:
Which will raise an `IndexError` already... so are you suggesting raising a new and specific exception to be handled by the caller?  I think, based on the API, that's better to return `None` to the caller, telling that "there's no latest task data".  What do you think?

Reply:
A `TypeError` could be raised here, if .get() returns `None`.

Reply:
I think `TypeError` will make it less clear way to say that there is no latest task data. Returning `None` could be a natural way and it could even be done without extra `try` clauses like:
```python
task_data = self._all_data.get(task_id)
if task_data is None:
    return None
else:
    return task_data[-1]
```

Reply:
I'm ok with that.

Reply:
> I think `TypeError` will make it less clear way to say that there is no latest task data. Returning `None` could be a natural way and it could even be done without extra `try` clauses like:
> 
> ```python
> task_data = self._all_data.get(task_id)
> if task_data is None:
>     return None
> else:
>     return task_data[-1]
> ```

Just a small optimization as there is no need to have an `else` here:

 ```python
 task_data = self._all_data.get(task_id)
 if task_data is None:
     return None
 return task_data[-1]
 ```